table_helper.cc

```diff
@@ -141,22 +141,22 @@ future<> table_helper::setup_keyspace(cql3::query_processor& qp, service::migrat
     opts["replication_factor"] = replication_factor;
     auto ksm = keyspace_metadata::new_keyspace(keyspace_name, "org.apache.cassandra.locator.SimpleStrategy", std::move(opts), true);
 
-    if (!db.has_keyspace(keyspace_name)) {
+    while (!db.has_keyspace(keyspace_name)) {
         auto group0_guard = co_await mm.start_group0_operation();
         auto ts = group0_guard.write_timestamp();
 
         if (!db.has_keyspace(keyspace_name)) {
-            co_await mm.announce(service::prepare_new_keyspace_announcement(db.real_database(), ksm, ts),
-                    std::move(group0_guard), format("table_helper: create {} keyspace", keyspace_name));
+            try {
+                co_await mm.announce(service::prepare_new_keyspace_announcement(db.real_database(), ksm, ts),
+                        std::move(group0_guard), format("table_helper: create {} keyspace", keyspace_name));
+            } catch (service::group0_concurrent_modification&) {
+            }
```

Comment:
I was following the logging practices in those places. `table_helper` doesn't log anything even in exceptional situations. It doesn't even have a logger. On the other hand, other places log information in similar situations.

Reply:
I see. I think we should go ahead and add a logger, then log a message about retry here like in other places. While in general we should eventually stop getting the `group0_concurrent_modification` exception, I think we might theoretically get stuck here due to a livelock because another node keeps doing group0 operations faster than us - having this message would let us know where we got stuck.